Keas.Mvc/Controllers/Api/EquipmentController.cs

```diff
@@ -324,16 +324,24 @@ public async Task<IActionResult> GetHistory(int id)
             return Json(history);
         }
 
-        public async Task<BigFixComputerProperties> GetComputer(string id)
+        public async Task<IActionResult> GetComputer(string id)
         {
-            return await this._bigfixService.GetComputer(id);
+            var result = await this._bigfixService.GetComputer(id);
```

Comment:
if you want to try/catch here, that would be a better spot.  Also, why not just handle it in the client?  is the appropriate data not being transferred through?

Reply:
so, as you mentioned, for now, I handled the exception here. one thing I am not quite sure is that how bigfix server handles the invalid id. if we don't catch the exception here, on our client-side we receive a 500 Internal Server Error.